astroquery/ukidss/core.py

```diff
@@ -1,48 +1,14 @@
 # Licensed under a 3-clause BSD style license - see LICENSE.rst
 from __future__ import print_function
 
-import warnings
-import re
-import time
-from math import cos, radians
-import requests
-from bs4 import BeautifulSoup
-
-from astropy.extern.six import BytesIO
-import astropy.units as u
-import astropy.coordinates as coord
-import astropy.io.votable as votable
-
-from ..query import QueryWithLogin
-from ..exceptions import InvalidQueryError, TimeoutError, NoResultsWarning
-from ..utils import commons
 from . import conf
-from ..exceptions import TableParseError
-
-__all__ = ['Ukidss', 'UkidssClass', 'clean_catalog']
-
-
-def validate_frame(func):
-    def wrapper(*args, **kwargs):
-        frame_type = kwargs.get('frame_type')
-        if frame_type not in UkidssClass.frame_types:
-            raise ValueError("Invalid frame type. Valid frame types are: {!s}"
-                             .format(UkidssClass.frame_types))
-        return func(*args, **kwargs)
-    return wrapper
 
+from ..wfau import BaseWFAUClass, clean_catalog
 
-def validate_filter(func):
-    def wrapper(*args, **kwargs):
-        waveband = kwargs.get('waveband')
-        if waveband not in UkidssClass.filters:
-            raise ValueError("Invalid waveband. Valid wavebands are: {!s}"
-                             .format(UkidssClass.filters.keys()))
-        return func(*args, **kwargs)
-    return wrapper
+__all__ = ['Ukidss', 'UkidssClass', 'clean_catalog']
```

Comment:
Is ``clean_catalog`` needed here, or it's enough to just import it into ``__init__.py`` directly from ``wfau``?

Reply:
I think it's only needed in `__init__`

Reply:
(ouch, meant to only comment in one batch, hold on modifying as I may have other minor comments).